styx-scheduler-service/src/main/java/com/spotify/styx/workflow/WorkflowInitializer.java

```diff
@@ -45,15 +45,7 @@ public WorkflowInitializer(Storage storage, Time time) {
     this.time = Objects.requireNonNull(time);
   }
 
-  public void inspectChange(Workflow workflow) {
-    final Optional<Workflow> previous;
-    try {
-      previous = storage.workflow(workflow.id());
-    } catch (IOException e) {
-      LOG.warn("failed to read workflow {} from storage", workflow.id(), e);
-      throw new RuntimeException(e);
-    }
-
+  public void inspectChange(Optional<Workflow> previous, Workflow workflow) {
```

Comment:
The lookup and mutation in this method should be in a transaction.

Reply:
I restored the code to make room for adding transaction which I prefer to do it in a separated PR.